### DISCUSSION TITLE
Connection issue on AWS Linux 2023

### DISCUSSION THREAD

**Discussion by fgreco55** [milvus-io/milvus#31287] [Q&A and General discussion]
Hi... I have successfully run Milvus on my iMac for a while now on a decent-sized Java RAG system.  Works great.   I'm now installing it in a cloud environment (AWS initially).  I'm having trouble connecting to the server.    Docker is running, and I see "milvus run standalone" when I do a ps.   I start milvus_cli using "sudo docker run -it zilliz/milvus_cli:latest".   I try to connect to the server with "connect", and I get: "Connect to Milvus error!<MilvusException: (code=2, message=Fail connecting to server on 127.0.0.1:19530. Timeout)>"     "telnet localhost 19530" seems to work; at least it's seemingly connecting.  

Any suggested tests or diagnostics to run are appreciated.

**Comment by yhmo** [comment]
Try using the physical IP to connect.

**Comment by fgreco55** [comment]
If you mean the "explicit" IP addr (IPs are logical), I already tried that.  Docker compose says I have everything running:

[+] Running 3/0
 ✔ Container milvus-etcd        Running    0.0s
 ✔ Container milvus-minio       Running    0.0s
 ✔ Container milvus-standalone  Running    0.0s

--------------------------------------------------------

% lsof -i -P -n | grep LISTEN | grep 19530
com.docke  3674 fgreco  233u  IPv6 0x556a1c228ec43aa1      0t0  TCP *:19530 (LISTEN)

--------------------------------------------------------

But no connection:

milvus_cli > connect
Connect to Milvus error!<MilvusException: (code=2, message=Fail connecting to server on 127.0.0.1:19530. Timeout)>
milvus_cli > connect
Connect to Milvus error!<MilvusException: (code=2, message=Fail connecting to server on 127.0.0.1:19530. Timeout)>

milvus_cli > connect -uri http://localhost:19530
Connect to Milvus error!<MilvusException: (code=2, message=Fail connecting to server on localhost:19530. Timeout)>
milvus_cli > connect -uri http://127.0.0.1:19530
Connect to Milvus error!<MilvusException: (code=2, message=Fail connecting to server on 127.0.0.1:19530. Timeout)>


Any suggestions?   Is there a server log somewhere that might shed some light?

**Comment by yhmo** [comment]
Use `docker logs [container_id]` to check the log of the milvus.
If the milvus is ready, there should be log like this:
[2024/03/20 02:24:49.560 +00:00] [DEBUG] [proxy/service.go:412] ["start Proxy server done"]
[2024/03/20 02:24:49.560 +00:00] [INFO] [components/proxy.go:58] ["Proxy successfully started"]

**Reply by fgreco55** [reply]
Are your docs up to date?  I'm getting a lot of 404s by following your installation pages.  wget hangs because it can't find files.

When you say "check the log of the milvus", what *exactly* do you mean?  Your yaml started 6 containers.  Which container is "the milvus"?  Is it the 'latest'?  etcd?  minio?  

I'm trying to teach RAG using Milvus.

**Reply by yhmo** [reply]
The latest version of v2.3 install page is here: https://milvus.io/docs/v2.3.x/install_standalone-docker-compose.md

This is the docker compose yaml to deploy a standalone milvus: `wget https://github.com/milvus-io/milvus/releases/download/v2.3.12/milvus-standalone-docker-compose.yml -O docker-compose.yml
`
After donwload this file, run this command to start the containers(docker and docker-compose are required):
`docker-compose up -d`

Three containers are launched:
![Screenshot 2024-03-20 234201](https://github.com/milvus-io/milvus/assets/2282099/3e7b19df-7897-4b62-96f1-c5b64f10c587)

If you use "docker ps" command, you will see the details, including id of each container:
![Screenshot 2024-03-20 234213](https://github.com/milvus-io/milvus/assets/2282099/cb1faada-4626-414e-b603-0f9df6a5cf27)

The three containers:
"milvus-etcd" ----------------- this is etcd server to store milvus metadata
"milvus-minio" --------------- this is minio server to store data files
"milvus-standalone"---------- this is milvus server

Use command "docker logs [container_id]", you will see the log of a container, replace the "container_id" by the milvus-standalone's id

**Reply by fgreco55** [reply]
on page: https://milvus.io/docs/v2.3.x/install_standalone-docker-compose.md
it says to get the standalone yaml file from the releases dir.   wget hangs for me, prob due to a 404.
I don't see a 'releases' dir when I explicitly visit that URL.

**Reply by fgreco55** [reply]
Maybe that URL should be: 
https://github.com/milvus-io/milvus/tree/master/deployments/docker/standalone

**Reply by fgreco55** [reply]
In the logs, I see:

[2024/03/20 16:34:42.215 +00:00] [DEBUG] [proxy/service.go:361] ["start Proxy server"]
...
[2024/03/20 16:34:42.220 +00:00] [INFO] [components/proxy.go:55] ["Proxy successfully started"]
...

milvus_cli > connect -uri http://127.0.0.1:19530
Connect to Milvus error!<MilvusException: (code=2, message=Fail connecting to server on 127.0.0.1:19530. Timeout)>
milvus_cli > connect -uri http://localhost:19530
Connect to Milvus error!<MilvusException: (code=2, message=Fail connecting to server on localhost:19530. Timeout)>

**Reply by fgreco55** [reply]
I finally got Milvus to start.  The typo in your docs was the problem. There is no 'releases' dir.

My app works, but milvus-cli still doesn't connect for some reason.

**Reply by yhmo** [reply]
Most of milvus users download the yaml file by this command, not sure why you got 404.

The link https://github.com/milvus-io/milvus/tree/master/deployments/docker/standalone is not an official release version. Here is a zip file for you, which contains the official 2.3.12 yaml file:
[docker-compose.zip](https://github.com/milvus-io/milvus/files/14681045/docker-compose.zip)
